pkg/lib/configreader/reader.go

```diff
@@ -1130,11 +1130,24 @@ func setField(val interface{}, destStruct interface{}, fieldName string) error {
 		debug.Ppg(destStruct)
 		return errors.Wrap(ErrorCannotSetStructField(), fieldName)
 	}
+
+	if val == nil {
```

Comment:
What is this for? Is this whole if block implemented because we now return `nil` interfaces from `DefaultDependentFieldsFunc`?

Reply:
Yes! In general, this block is necessary to set fields in a struct to nil (which I guess we had never done, but now we do in the `DefaultDependentFieldsFunc` I added)